### ISSUE TITLE
Http.RequestBuilder is double-decoding query parameters

### ISSUE THREAD

**Issue by hamiltron** [playframework/playframework#6229]
### Expected Behavior

When using `RequestBuilder` in controller unit tests, the request object generated and passed to the tested controller should have query parameters that are URL-decoded once. 

### Actual Behavior

The request object query parameters are URL-decoded twice, resulting in '%2B' being decoded as ' ' rather than '+'. 

### Reproducible Test Case

```
System.err.println(new Http.RequestBuilder().uri("path?query=x%2By").build().getQueryString("query"));
```

Result is `x y`, it should be `x+y`

### Suggested Fix

[Http.RequestBuilder.splitQuery](https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/java/play/mvc/Http.java#L1349)
should use [java.net.URI.getRawQuery](http://docs.oracle.com/javase/7/docs/api/java/net/URI.html#getRawQuery())

**Comment by gmethvin** [comment]
@hamiltron Thanks for finding this. Just submitted a fix: #6230

**Comment by hamiltron** [comment]
Great! Thanks for jumping on it. I looked for a release schedule but I didn't find anything. Do you have any notion of when we would see this backported and released for 2.4.x?

**Comment by etleap-nathan** [comment]
It'd be great if this were backported for 2.4.x.
